### PR TITLE
Migrate ethers side dependencies to v6

### PR DESCRIPTION
**Pull request by MCarlomagno**
Adapted relay-signer-client to ethers v6 

Fixes PLAT-4046

### CODE REVIEW

File: packages/relay-signer/src/relayer.ts [hunk 1]
```diff
@@ -43,9 +43,13 @@ export class Relayer implements IRelayer {
...
 
-  public getSigner(provider: Provider, options: DefenderRelaySignerOptions = {}): DefenderRelaySigner {
+  public async getSigner(
+    provider: JsonRpcProvider,
+    options: DefenderRelaySignerOptions = {},
+  ): Promise<DefenderRelaySigner> {
     if (!this.credentials) throw new Error(`Missing credentials for creating a DefenderRelaySigner instance.`);
-    return new DefenderRelaySigner(this.credentials, provider, options);
+    const relayer = await this.relayer.getRelayer();
+    return new DefenderRelaySigner(this.credentials, provider, relayer.address, options);
```
Comment (MCarlomagno) [inline packages/relay-signer/src/relayer.ts:52]: Ethers v6 signer requires also the sender address

File: packages/relay-signer/src/ethers/signer.ts [hunk 2]
```diff
@@ -218,22 +233,22 @@ export class DefenderRelaySigner extends Signer implements TypedDataSigner {
...
-        logger.throwArgumentError('invalid transaction key: ' + key, 'transaction', transaction);
+        console.error('invalid transaction key: ' + key, 'transaction', transaction);
       }
     }
-    const tx = shallowCopy(transaction);
+    const tx: Deferrable<DefenderTransactionRequest> = { ...transaction };
 
     tx.from = Promise.all([Promise.resolve(tx.from), this.getAddress()]).then((result) => {
       if (!!result[0] && result[0].toLowerCase() !== result[1].toLowerCase()) {
-        logger.throwArgumentError('from address mismatch', 'transaction', transaction);
```
Comment (MCarlomagno) [inline packages/relay-signer/src/ethers/signer.ts:236]: The ehters logger was deprecated after v5

File: packages/relay-signer/src/ethers/signer.ts [hunk 3]
```diff
@@ -1,21 +1,31 @@
...
-import { Transaction } from '@ethersproject/transactions';
 import { omit } from 'lodash';
 import { Speed } from '../models/transactions';
 import { RelayerParams } from '../models/relayer';
 import { isEIP1559Tx, isLegacyTx, isRelayer } from './utils';
 
-const logger = new Logger(`@openzeppelin/defender-relay-client`);
+export type Deferrable<T> = {
+  [K in keyof T]: T[K] | Promise<T[K]>;
+};
```
Comment (MCarlomagno) [inline packages/relay-signer/src/ethers/signer.ts:28]: The code was deleted after ethers v5 and we need it for internal use

**Review by zeljkoX — DISMISSED** [review]
LGTM

Lets ensure that we test this throughly

File: packages/monitor/package.json [hunk 4]
```diff
@@ -20,8 +20,8 @@
   "author": "OpenZeppelin Defender <defender@openzeppelin.com>",
   "license": "MIT",
   "dependencies": {
-    "@ethersproject/abi": "^5.7.0",
-    "@openzeppelin/defender-sdk-base-client": "^1.9.0",
+    "ethers": "^6.9.0",
+    "@openzeppelin/defender-sdk-base-client": "^1.8.0",
```
[the remark this answers is not in the document]
Comment (MCarlomagno) [reply packages/monitor/package.json:24]: Good catch, I probably missed that when I ran `pnpm install`

**Review by zeljkoX — APPROVED** [review]

**Pull request merged**